Source/JavaScriptCore/dfg/DFGStoreBarrierInsertionPhase.cpp

```diff
@@ -431,17 +435,50 @@ class StoreBarrierInsertionPhase : public Phase {
                 // So this analysis must mark an allocation that escapes to the heap as being part of the primordial
                 // epoch.
 
-                ClobberSet writes = writeSet(m_graph, m_node);
-                if (writes.overlaps(Heap) || writes.overlaps(Stack)) {
-                    auto escape = [] (Node* node) {
+                auto readFunc = [&] (const AbstractHeap& heap) {
+                    if (heap.kind() == Stack) {
+                        Vector<AbstractHeap, 4> toRemove;
+                        for (auto [trackedHeap, node] : potentialStackEscapes) {
+                            if (trackedHeap.overlaps(heap)) {
+                                node->setEpoch(Epoch());
+                                toRemove.append(trackedHeap);
+                            }
+                        }
+
+                        for (const auto& heap : toRemove)
+                            potentialStackEscapes.remove(heap);
+                    }
+                };
+
+                bool wroteHeapOrStack = false;
+                AbstractHeap preciseStackWrite;
+                auto writeFunc = [&] (const AbstractHeap& heap) {
+                    if (heap.kind() == Stack) {
+                        wroteHeapOrStack = true;
+                        if (!heap.payload().isTop())
+                            preciseStackWrite = heap;
```

Comment:
Can it be a Vector?
For example, `LoadVarargs` / `ForwardVarargs` are doing `write(AbstractHeap(Stack, ...))` multiple times.

Reply:
Yeah, but we're only considering this for `PutStack`/`SetLocal`. Let me add a RELEASE_ASSERT that for these nodes we only see a single stack write.